src/main/java/uk/gov/hmcts/reform/slc/services/UpdateLetterStatusJob.java

```diff
@@ -42,7 +42,7 @@ public void run() {
                 .map(parser::parse)
                 .forEach(parsedReport -> {
                     parsedReport.statuses.forEach(sendLetterClient::updatePrintedAt);
```

Comment:
Given we've successfully parsed the file, maybe a single record fail shouldn't stop the whole file processing? We could just fail for that record. If there are thousands of records in a file, it's quite likely that at some point we'll get an error response for some reason.

Not for this pull request, but it would be worth retrying on connection-related errors, timeouts and maybe server errors.

Reply:
What happens when current method throws exception?